### PR TITLE
adds url param support to code and resources pages.

### PR DESCRIPTION
**Pull request by kevmodrome**
* URL params are now updated when you select and unselect tags.
* Code and Resources pages now support linking directly to tags like so:

`.../resources?tag=official&tag=web+app+frameworks`

### CODE REVIEW

**Comment by swyxio** [comment]
Deploy Preview is done!

Built with commit 81c2f1af442a8fcf39df926334e5828822442331

https://deploy-preview-100--svelte-community.netlify.com

**Comment by kevmodrome** [comment]
This solves #25

File: src/routes/code.svelte [hunk 1]
```diff
@@ -29,15 +30,41 @@
...
     if (selectedTags.has(tag)) {
       selectedTags.delete(tag);
     } else {
       selectedTags.add(tag);
     }
     selectedTags = selectedTags; // TODO maybe use immutable patterns instead?
+    updateUrl(selectedTags);
     scrollToTop();
   };
   const clearSelectedTags = () => {
```
Comment (ryanatkn) [inline src/routes/code.svelte:70]: In `clearSelectedTags` we want to call `updateUrl(selectedTags)` after reassigning `selectedTags = selectedTags;` like we do in `toggleTag`.

File: src/routes/resources.svelte [hunk 2]
```diff
@@ -31,13 +44,27 @@
...
     if (selectedTags.has(tag)) {
       selectedTags.delete(tag);
     } else {
       selectedTags.add(tag);
     }
     selectedTags = selectedTags; // TODO maybe use immutable patterns instead?
+    updateUrl(selectedTags);
     scrollToTop();
   };
   const clearSelectedTags = () => {
```
Comment (ryanatkn) [inline src/routes/resources.svelte:70]: In `clearSelectedTags` we want to call `updateUrl(selectedTags)` after reassigning `selectedTags = selectedTags;` like we do in `toggleTag`.

**Review by ryanatkn — CHANGES_REQUESTED** [review]
Yay thank you, this is awesome! I was just wanting to link to a specific tag.

I saw one issue where the tags aren't updated when the tags are cleared. The fix I suggested (see inline comments) does leave a `?` in the address bar, might be easy to fix.

Another minor thing we can fix later is that it filters after mounting, so you get a flash of full content, which is no problem but could be polished. I'll probably look for a way to fix it if nobody beats me to it. I see no reason to fix in this PR unless that's what you'd prefer to do.

**Comment by kevmodrome** [comment]
Makes sense, i'll update that and push a new commit

**Review by ryanatkn — APPROVED** [review]

**Pull request merged**